korangar/src/loaders/animation/mod.rs

```diff
@@ -224,59 +222,55 @@ impl AnimationLoader {
             let mut min_left = i32::MAX;
             let mut max_right = 0;
 
-            // The player can change the sprite and causes an offset from a image to
-            // another.
-            if entity_type == EntityType::Player {
-                // Create a bounding box to standardize the size of the player sprite, ensuring
-                // that every sprite has the same proportion and that pixels are rendered in
-                // the correct position.
-                // The player sprite is generally 110 pixels, with an additional 40 pixels added
-                // for extra space, making the total height 150 pixels.
-                let extra_size = 20;
-                min_top = -90;
-                max_bottom = 20;
-                min_top -= extra_size;
-                max_bottom += extra_size;
-                min_left = -50;
-                max_right = 50;
-            } else {
-                animations[action_index].frames.iter().for_each(|frame| {
-                    let center_x = (frame.size.x - 1) / 2;
-                    min_left = min(min_left, frame.offset.x - center_x);
-                    max_right = max(max_right, frame.offset.x + (frame.size.x - 1 - center_x));
-
-                    let center_y = (frame.size.y - 1) / 2;
-                    min_top = min(min_top, frame.offset.y - center_y);
-                    max_bottom = max(max_bottom, frame.offset.y + (frame.size.y - 1 - center_y));
-                });
-            }
+            animations[action_index].frames.iter().for_each(|frame| {
+                let center_x = (frame.size.x - 1) / 2;
+                min_left = min(min_left, frame.offset.x - center_x);
+                max_right = max(max_right, frame.offset.x + (frame.size.x - 1 - center_x));
+
+                let center_y = (frame.size.y - 1) / 2;
+                min_top = min(min_top, frame.offset.y - center_y);
+                max_bottom = max(max_bottom, frame.offset.y + (frame.size.y - 1 - center_y));
+            });
 
+            // From here, the code start not using pixel coordinates (center of pixel

```

Comment:
Can you explain in the comment why we do this coordinate shift?

Reply:
Yeah, I explained better in the code, but the main reason is that you need the pixel border during rendering process.